python/pyspark/shell.py

```diff
@@ -45,11 +45,29 @@
         if parent_dir in sys.path:
             sys.path.remove(parent_dir)
 
-
 if is_remote():
     try:
         # Creates pyspark.sql.connect.SparkSession.
         spark = SparkSession.builder.getOrCreate()
```

Comment:
Could we get this from `spark.conf`? I think we should avoid relying on environment variable in general. Should probably also disable `spark.ui.showConsoleProgress` here.